cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/BrokerFailureDetector.java

```diff
@@ -113,7 +118,15 @@ synchronized void detectBrokerFailures(boolean skipReportingIfNotUpdated) {
   }
 
   void shutdown() {
-    _zkClient.close();
+    _detectionExecutor.shutdown();
+    try {
+      _detectionExecutor.awaitTermination(2, TimeUnit.MINUTES);
```

Comment:
Nit: `2 minutes` might be a little long for the timeout. Would it be possible to limit this to 30 seconds? If we are already shutting down CC, would it be ok to await for a shorter time to detect potential broker failures?